client/test/fabricClientConnection.test.ts

```diff
@@ -0,0 +1,156 @@
+            peer.getName().should.deep.equal('peerTwo');
+        });
+    });
+
+    describe('getAllChannelsForPeer', () => {
+        it('should get all the channels a peer has joined', async () => {
+            const peerOne: fabricClient.Peer = new fabricClient.Peer('grpc://localhost:1454', {name: 'peerOne'});
+
+            fabricClientStub.getPeersForOrg.returns([peerOne]);
+
+            const channelOne = {channel_id: 'channel-one'};
+            const channelTwo = {channel_id: 'channel-two'};
+            fabricClientStub.queryChannels.resolves({channels: [channelOne, channelTwo]});
+
+            await fabricClientConnection.connect();
+
+            const channelNames: Array<string> = await fabricClientConnection.getAllChannelsForPeer('peerTwo');
+
+            channelNames.should.deep.equal(['channel-one', 'channel-two']);
+        });
+    });
+
+    describe('getInstalledChaincode', () => {
+        it('should get the install chaincode', async () => {
+            const peerOne: fabricClient.Peer = new fabricClient.Peer('grpc://localhost:1454', {name: 'peerOne'});
+            const peerTwo: fabricClient.Peer = new fabricClient.Peer('grpc://localhost:1453', {name: 'peerTwo'});
+
+            fabricClientStub.getPeersForOrg.returns([peerOne, peerTwo]);
+
+            fabricClientStub.queryInstalledChaincodes.withArgs(peerOne).resolves({
+                chaincodes: [{
+                    name: 'biscuit-network',
+                    version: '0.7'
+                }, {name: 'biscuit-network', version: '0.8'}, {name: 'cake-network', version: '0.8'}]
+            });
+
+            await fabricClientConnection.connect();
+            const installedChaincode: Map<string, Array<string>> = await fabricClientConnection.getInstalledChaincode('peerOne');
+            installedChaincode.size.should.equal(2);
+            Array.from(installedChaincode.keys()).should.deep.equal(['biscuit-network', 'cake-network']);
+            installedChaincode.get('biscuit-network').should.deep.equal(['0.7', '0.8']);
+            installedChaincode.get('cake-network').should.deep.equal(['0.8']);
+        });
+    });
+
+    describe('getInstantiatedChaincode', () => {
+        it('should get the instantiated chaincode', async () => {
+            const channelOne = {channel_id: 'channel-one', queryInstantiatedChaincodes: mySandBox.stub()};
+
+            channelOne.queryInstantiatedChaincodes.resolves({
+                chaincodes: [{
+                    name: 'biscuit-network',
+                    version: '0,7'
+                }, {name: 'cake-network', version: '0.8'}]
+            });
+
+            fabricClientStub.getChannel.returns(channelOne);
+
+            await fabricClientConnection.connect();
+            const instantiatedChainodes: Array<any> = await fabricClientConnection.getInstantiatedChaincode('channel-one');
+
+            instantiatedChainodes.should.deep.equal([{name: 'biscuit-network', version: '0,7'}, {
```

Comment:
is there a reason the biscuit-network entry is on one line and cake-network is split across two?
I'm being picky :)

Reply:
fixed